packages/sil-governance/src/store/index.ts

```diff
@@ -1,12 +1,59 @@
-import { applyMiddleware, compose, createStore, Middleware, Store } from "redux";
+import {
+  AnyAction,
+  applyMiddleware,
+  compose,
+  createStore,
+  Dispatch,
+  Middleware,
+  MiddlewareAPI,
+  Store,
+} from "redux";
 
+import { getProposals, replaceProposalsAction, ReplaceProposalsActionType } from "./proposals";
 import reducer, { RootReducer, RootState } from "./reducers";
 
 const composeEnhancers =
   (typeof window === "object" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || // eslint-disable-line
   compose;
 
-const middlewares: readonly Middleware[] = [];
+const replaceProposals = async (
+  dispatch: Dispatch<ReplaceProposalsActionType>,
+  state: RootState,
+): Promise<void> => {
+  const governor = state.extension.governor;
+
+  if (governor) {
+    const proposals = await getProposals(governor);
+    dispatch(replaceProposalsAction(proposals));
+  }
+};
+
+const refreshProposalsActionMw = ({
+  dispatch,
+  getState,
+}: MiddlewareAPI): ((next: Dispatch<AnyAction>) => (action: AnyAction) => any) => next => {
+  return action => {
+    if (action.type === "@@proposals/REFRESH") replaceProposals(dispatch, getState());
+
+    next(action);
+  };
+};
+
+const refreshProposalsPeriodicallyMw: Middleware<{}, RootState, Dispatch<AnyAction>> = ({
+  dispatch,
+  getState,
+}: MiddlewareAPI): ((next: Dispatch<AnyAction>) => (action: AnyAction) => any) => next => {
+  const refreshProposals = async (): Promise<void> => {
+    await replaceProposals(dispatch, getState());
+    setTimeout(refreshProposals, 3000);
+  };
+
+  refreshProposals();
```

Comment:
This will create timeout handler on every update. I am not sure so please correct me if I am wrong.

Reply:
I thought about that but I saw several implementations using it, so not sure.

Reply:
ok, but can you check this please?